docs/src/proposals/timely-vote-credits.md

```diff
@@ -0,0 +1,136 @@
+vote should be.  Assuming linear reduction in vote credits awarded (where 1
+slot of additional lag reduces earned vote credits by 1), the maximum vote
+credits value determines how much "penalty" there is for each additional slot
+of latency.  For example, a value of 10 would mean that after the grace period
+slots, every additional slot of latency would result in a 10% reduction in
+vote credits earned as each subsequent slot earns 1 credit less out of a
+maximum possible 10 credits.
+
+Again, historical voting data was analyzed over many epochs and the conclusion
+drawn was that a maximum credits of 10 is the largest value that can be used
+and still have a noticeable effect on known laggers.  Values higher than that
+result in such a small penalty for each slot of lagging that intentional
+lagging is still too profitable.  Lower values are even more punishing to
+intentional lagging; but an attempt has been made to conservatively choose the
+highest value that produces noticeable results.
+
+The selection of these values is partially documented here:
+
+https://www.shinobi-systems.com/timely_voting_proposal
+
+The above document is somewhat out of date with more recent analysis, which
+occurred in this github issue:
+
+https://github.com/solana-labs/solana/issues/19002
+
+### Method of Implementation
+
+When a Vote or VoteStateUpdate instruction is received by a validator, it will
+use the Clock sysvar to identify the slot in which that instruction has
+landed.  For any newly voted on slot within that Vote or VoteStateUpdate
+transaction, the validator will record the vote latency of that slot as
+(voted_in_slot - voted_on_slot).  It will include safety mechanisms in case
+there are bugs; if voted_on_slot is greater than or equal to voted_in_slot, a
+vote latency of 1 will be computed instead of the large value that would
+result from integer subtraction underflow.
+
+These vote latencies will be stored a new vector of u8 latency values appended
+to the end of the VoteState.  VoteState currently has ~200 bytes of free space
+at the end that is unused, so this new vector of u8 values should easily fit
+within this available space.  Because VoteState is an ABI frozen structure,
+utilizing the mechanisms for updating frozen ABI will be required, which will
+complicate the change.  Furthermore, because VoteState is embedded in the
+Tower data structure and it is frozen ABI as well, updates to the frozen ABI
+mechanisms for Tower will be needed also.  These are almost entirely
+mechanical changes though, that involve ensuring that older versions of these
+data structures can be updated to the new version as they are read in, and the
+new version written out when the data structure is next persisted.
+
+The credits to award for a rooted slot will be calculated using the latency
+value stored in latency vector for the slot, and a formula that awards
+latencies of 1 - 3 ten credits, with a 1 credit reduction for each vote
```

Comment:
nit: `1 - 3` would be clearer spelled out and with a unit of measurement, i.e.

`1 to 3 slots`